feat: add resource okta_group_owners

exitcode0 opened the pull request:
fixes #2110

exitcode0 committed:
feat(idaas): add okta_group_owners

aditya-okta committed:
Merge branch 'master' into okta_group_owners

aditya-okta commented on okta/services/idaas/resource_okta_group_owners.go:
```diff
@@ -0,0 +1,430 @@
+		var err2 error
+		apiResp, err2 = apiResp.Next(&next)
+		if err2 != nil {
+			return result, nil // return what we have
```
why are we not returning the error here?

exitcode0 committed:
Merge branch 'okta:master' into okta_group_owners

exitcode0 committed:
fix(group_owners): harden authoritative behavior, improve error handling, extract helpers

  okta_group_owners (plural, authoritative):
  - Remove track_all_owners; resource is now always authoritative
  - Create: reconcile pre-existing owners by removing undeclared ones
  - Update: diff against live API instead of Terraform state
  - Delete: query live API instead of state, continue on individual errors
  - Write partial state on Create/Update failures for recovery
  - Remove SizeAtLeast(1) validator to allow declaring zero owners
  - Add ID prefix validation warning (00u=USER, 00g=GROUP) at plan time
  - Enrich 404 errors with prefix-mismatch hints while preserving raw API error
  - Fix errors.As target (use *okta.GenericOpenAPIError, not value type)
  - Add nil guard on apiResp.Response before accessing StatusCode
  - Normalize owner type to uppercase, detect duplicate owner IDs
  - Propagate pagination errors instead of swallowing them
  - Use SuppressErrorOn404_V5 consistently in listAllGroupOwners

  okta_group_owner (singular, non-authoritative):
  - Add RequiresReplace on group_id, id_of_group_owner, and type
  - Replace dead Update with error guard (API has no update endpoint)
  - Fix RemoveResource on transient Read errors destroying state on 500s
  - Add nil guard on apiResp.Response in Read
  - Set IdOfGroupOwner in mapGroupOwnerToState

  Shared:
  - Extract helpers to group_owner_helpers.go (isAlreadyAssignedOwnerError,
    warnIfIDTypeMismatch, assignOwnerErrorDetail, safeString, apiErrorBody)
  - Add SKU detection framework (acctest.RequireSKU, config.OktaSKU)
  - Register OktaIDaaSGroupOwners constant in resources.go
  - Regenerate docs via tfplugindocs

  Tests:
  - Add SKU gating and CheckDestroy to all tests
  - Rename _deletedGroup404 -> _basicCreateDestroy
  - Add group-as-owner and type-mismatch test cases
  - Re-record all VCR cassettes

exitcode0 committed:
fix(docs): restore group_owner.md to match upstream

exitcode0 committed:
fix(docs): update fallback templates for tfplugindocs v0.17+

  The printf two-pass rendering trick (outputting literal {{tffile}} and
  {{codefile}} directives for a second render pass) was a bug that
  tfplugindocs v0.17.0 intentionally removed. With v0.24.0 these
  directives are written to the final markdown as-is, leaking absolute
  local file paths.

  Switch to calling tffile and codefile as template functions directly,
  matching the pattern used by the per-resource templates.

  Note: one empty example file (data-sources/okta_api_service_integration/
  data-source.tf) will cause go generate to fail until it is given content
  or removed.

exitcode0 committed:
fix(docs): add per-resource templates for okta_group_owner(s)

aditya-okta commented on okta/services/idaas/resource_okta_group_owners.go:
```diff
@@ -0,0 +1,407 @@
+				continue
+			}
+			resp.Diagnostics.AddError("failed to delete group owner", fmt.Sprintf("group_id=%s owner_id=%s error=%s", groupID, o.ID, err))
+			continue
```
If one delete fails, Terraform marks the apply as errored. The remaining deletes then run but their results are ignored. If any subsequent delete also fails, only the first error is surfaced. Consider collecting all errors and returning after the loop, or stopping on first error.

Something like this would work 

```if err != nil && utils.SuppressErrorOn404_V5(apiResp, err) != nil {
			resp.Diagnostics.AddError(
				"failed to delete group owner",
				fmt.Sprintf("group_id=%s owner_id=%s error=%s", groupID, o.ID, err),
			)
```

aditya-okta committed:
add changes to report errors for every failed deletion by removing the redundant continue

exitcode0 committed:
fix: guard nil embedded http.Response in V5/V6 API helpers, re-record VCR

SuppressErrorOn404_V5/V6 and ResponseErr_V5/V6 access promoted fields
(StatusCode, Status) from the embedded *http.Response in APIResponse.
When the embedded pointer is nil — e.g. during VCR playback with a
stale cassette — this causes a nil pointer dereference panic.

Re-record the crudAndUpdate VCR cassette to include the additional
listAllGroupOwners call added by the ImportState hydration in 51895e6c.

dhiwakar-okta reviewed (approved):
LGTM 🚀
